### ISSUE TITLE
Builtin functions not considered intrinsic

### ISSUE THREAD

**Issue by ghost** [flang-compiler/flang#361]
Consider [this file](https://github.com/pyne/pyne/blob/5045633920885e9bae4840dddf38f3c54a0dd790/src/ensdf_processing/RULER/ruler.f).

It fails with the following error:

```
F90-S-0126-Name len_trim is not an intrinsic function (ruler.f: 714)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 715)
F90-S-0126-Name len is not an intrinsic function (ruler.f: 715)
F90-S-0126-Name max0 is not an intrinsic function (ruler.f: 715)
F90-S-0126-Name min0 is not an intrinsic function (ruler.f: 715)
F90-S-0126-Name dabs is not an intrinsic function (ruler.f: 718)
F90-S-0126-Name len_trim is not an intrinsic function (ruler.f: 1391)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 1391)
F90-S-0126-Name len is not an intrinsic function (ruler.f: 1391)
F90-S-0126-Name max0 is not an intrinsic function (ruler.f: 1391)
F90-S-0126-Name dsqrt is not an intrinsic function (ruler.f: 1392)
F90-S-0126-Name len_trim is not an intrinsic function (ruler.f: 1542)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 1542)
F90-S-0126-Name len is not an intrinsic function (ruler.f: 1542)
F90-S-0126-Name max0 is not an intrinsic function (ruler.f: 1542)
F90-S-0126-Name dabs is not an intrinsic function (ruler.f: 1544)
F90-S-0126-Name dsqrt is not an intrinsic function (ruler.f: 1544)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 1706)
F90-S-0126-Name dabs is not an intrinsic function (ruler.f: 1707)
F90-S-0126-Name dsqrt is not an intrinsic function (ruler.f: 1707)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 1896)
F90-S-0126-Name len is not an intrinsic function (ruler.f: 1927)
F90-S-0126-Name len_trim is not an intrinsic function (ruler.f: 1949)
F90-S-0126-Name len_trim is not an intrinsic function (ruler.f: 2004)
F90-S-0126-Name index is not an intrinsic function (ruler.f: 2039)
F90-F-0008-Error limit exceeded (ruler.f: 2039)
F90/x86-64 Windows Flang - 1.5 2017-05-01: compilation aborted
```

**Comment by xoviat** [comment]
Note that gfortran compiles this file cleanly.

**Comment by gklimowicz** [comment]
Should be fixed in recent Flang updates.

**Comment by gklimowicz** [comment]
Might not yet be in Flang yet; reopen until I can confirm.

**Comment by gklimowicz** [comment]
Fixed in pull request #458 commit 295d29a.
